backend/prompt_studio/prompt_studio_index_manager/models.py

```diff
@@ -90,31 +91,40 @@ class Meta:
         ]
 
 
+def delete_from_vector_db(index_ids_history, vector_db_instance_id):
+    org_schema = connection.tenant.schema_name
+    util = PromptIdeBaseTool(log_level=LogLevel.INFO, org_id=org_schema)
+    vector_db = VectorDB(
+        tool=util,
+        adapter_instance_id=vector_db_instance_id,
+    )
+    for index_id in index_ids_history:
+        logger.debug(f"Deleting from VectorDB - index id: {index_id}")
+        try:
+            vector_db.delete(ref_doc_id=index_id)
+        except Exception as e:
+            # Log error and continue with the next index id
+            logger.error(f"Error deleting index: {index_id} - {e}")
+
+
 # Function will be executed every time an instance of IndexManager is deleted.
-@receiver(post_delete, sender=IndexManager)
+@receiver(pre_delete, sender=IndexManager)
 def perform_vector_db_cleanup(sender, instance, **kwargs):
     """Signal to perform vector db cleanup."""
     logger.info("Performing vector db cleanup")
-    logger.debug("Document tool id: %s", instance.document_manager.tool_id)
+    logger.debug(f"Document tool id: {instance.document_manager.tool_id}")
     try:
         # Get the index_ids_history to clean up from the vector db
         index_ids_history = json.loads(instance.index_ids_history)
         vector_db_instance_id = str(instance.profile_manager.vector_store.id)
-        # Generate a run_id
-        org_schema = connection.tenant.schema_name
-        util = PromptIdeBaseTool(log_level=LogLevel.INFO, org_id=org_schema)
-        vector_db = VectorDB(
-            tool=util,
-            adapter_instance_id=vector_db_instance_id,
-        )
-        for index_id in index_ids_history:
-            logger.debug(f"Deleting from VectorDB - index id: {index_id}")
-            try:
-                vector_db.delete(ref_doc_id=index_id)
-            except Exception as e:
-                # Log error and continue with the next index id
-                logger.error(f"Error deleting index: {index_id} - {e}")
-    # Not raising any exception.
-    # Cleanup should not fail the deletion of the index manager.
+        delete_from_vector_db(index_ids_history, vector_db_instance_id)
     except SdkError as e:
```

Comment:
@athul-rs  Let's not catch specific exceptions and log them differently along with a catch-all. A single catch all that raises a warning and logs the error should do. I assume error logs are for scenarios where the user receives a 4xx or 5xx.
Also make sure to follow the convention of `raise error from e` to preserve the traceback
CC: @hari-kuriakose

Reply:
My bad, this is irrelevant here. I got confused while looking at a different piece of code